Expression extractor fails when there are properties to set

### Bug description

When we have a consumer endpoint with query parameters, the expression extractor fails with an NPE of type:
```
org.apache.camel.FailedToCreateRouteException: Failed to create route route5: Route(route5)[From[timer:cTimer_1?period=1&repeatCount=1&del... because of Failed to resolve endpoint: timer://cTimer_1?delay=1&period=1&repeatCount=1 due to: Cannot invoke "org.apache.camel.Component.getEndpointPropertyConfigurer()" because the return value of "org.apache.camel.support.DefaultEndpoint.getComponent()" is null
	at org.apache.camel.reifier.RouteReifier.createRoute(RouteReifier.java:82)
	at org.apache.camel.impl.DefaultModelReifierFactory.createRoute(DefaultModelReifierFactory.java:49)
	at org.apache.camel.impl.DefaultCamelContext.startRouteDefinitions(DefaultCamelContext.java:816)
	at org.apache.camel.impl.DefaultCamelContext.startRouteDefinitions(DefaultCamelContext.java:685)
	at org.apache.camel.impl.engine.AbstractCamelContext.doInit(AbstractCamelContext.java:2396)
	at org.apache.camel.support.service.BaseService.init(BaseService.java:83)
	at org.apache.camel.impl.engine.AbstractCamelContext.init(AbstractCamelContext.java:2062)
	at org.apache.camel.support.service.BaseService.start(BaseService.java:111)
	at org.apache.camel.impl.engine.AbstractCamelContext.start(AbstractCamelContext.java:2081)
	at org.apache.camel.impl.DefaultCamelContext.start(DefaultCamelContext.java:313)
	at org.apache.camel.quarkus.support.language.deployment.dm.DryModeMain.doStart(DryModeMain.java:81)
	at org.apache.camel.support.service.BaseService.start(BaseService.java:119)
	at org.apache.camel.quarkus.support.language.deployment.LanguageSupportProcessor.extractExpressions(LanguageSupportProcessor.java:62)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:909)
	at io.quarkus.builder.BuildContext.run(BuildContext.java:282)
	at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2513)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1538)
	at java.base/java.lang.Thread.run(Thread.java:833)
	at org.jboss.threads.JBossThread.run(JBossThread.java:501)
Caused by: org.apache.camel.ResolveEndpointFailedException: Failed to resolve endpoint: timer://cTimer_1?delay=1&period=1&repeatCount=1 due to: Cannot invoke "org.apache.camel.Component.getEndpointPropertyConfigurer()" because the return value of "org.apache.camel.support.DefaultEndpoint.getComponent()" is null
	at org.apache.camel.impl.engine.AbstractCamelContext.doGetEndpoint(AbstractCamelContext.java:856)
	at org.apache.camel.impl.engine.AbstractCamelContext.getEndpoint(AbstractCamelContext.java:758)
	at org.apache.camel.support.CamelContextHelper.getMandatoryEndpoint(CamelContextHelper.java:60)
	at org.apache.camel.reifier.AbstractReifier.resolveEndpoint(AbstractReifier.java:195)
	at org.apache.camel.reifier.RouteReifier.doCreateRoute(RouteReifier.java:96)
	at org.apache.camel.reifier.RouteReifier.createRoute(RouteReifier.java:76)
	... 23 common frames omitted
Caused by: java.lang.NullPointerException: Cannot invoke "org.apache.camel.Component.getEndpointPropertyConfigurer()" because the return value of "org.apache.camel.support.DefaultEndpoint.getComponent()" is null
	at org.apache.camel.support.DefaultEndpoint.setProperties(DefaultEndpoint.java:414)
	at org.apache.camel.support.DefaultEndpoint.configureProperties(DefaultEndpoint.java:394)
	at org.apache.camel.support.DefaultComponent.setProperties(DefaultComponent.java:418)
	at org.apache.camel.support.DefaultComponent.createEndpoint(DefaultComponent.java:186)
	at org.apache.camel.impl.engine.AbstractCamelContext.doGetEndpoint(AbstractCamelContext.java:822)
	... 28 common frames omitted
```
With a route of type:
```
from("timer:myTimer?period=1&repeatCount=1&delay=1")
               .log("body value is: ${body}");
```

--- a/extensions-support/language/deployment/pom.xml
+++ b/extensions-support/language/deployment/pom.xml
@@ -42,6 +42,16 @@
             <groupId>org.apache.camel</groupId>
             <artifactId>camel-main</artifactId>
         </dependency>
+        <dependency>
+            <groupId>org.apache.camel</groupId>
+            <artifactId>camel-timer</artifactId>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>io.quarkus</groupId>
+            <artifactId>quarkus-junit5-internal</artifactId>
+            <scope>test</scope>
+        </dependency>
     </dependencies>
 
     <build>

--- a/extensions-support/language/deployment/src/main/java/org/apache/camel/quarkus/support/language/deployment/dm/DryModeComponent.java
+++ b/extensions-support/language/deployment/src/main/java/org/apache/camel/quarkus/support/language/deployment/dm/DryModeComponent.java
@@ -31,6 +31,13 @@ public class DryModeComponent extends DefaultComponent {
 
     @Override
     protected Endpoint createEndpoint(String uri, String remaining, Map<String, Object> parameters) {
-        return new DryModeEndpoint(uri);
+        DryModeEndpoint endpoint = new DryModeEndpoint(uri);
+        endpoint.setComponent(this);
+        return endpoint;
+    }
+
+    @Override
+    protected boolean resolveRawParameterValues() {
+        return false;
     }
 }

--- a/extensions-support/language/deployment/src/main/java/org/apache/camel/quarkus/support/language/deployment/dm/DryModeEndpoint.java
+++ b/extensions-support/language/deployment/src/main/java/org/apache/camel/quarkus/support/language/deployment/dm/DryModeEndpoint.java
@@ -55,6 +55,11 @@ public class DryModeEndpoint extends DefaultEndpoint {
         return String.format("dm:%s", uri);
     }
 
+    @Override
+    public boolean isLenientProperties() {
+        return true;
+    }
+
     private static class DryModeConsumer extends DefaultConsumer {
 
         DryModeConsumer(Endpoint endpoint, Processor processor) {

--- a/extensions-support/language/deployment/src/test/java/org/apache/camel/quarkus/support/language/deployment/PropertiesToSetTest.java
+++ b/extensions-support/language/deployment/src/test/java/org/apache/camel/quarkus/support/language/deployment/PropertiesToSetTest.java
@@ -1,0 +1,82 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.camel.quarkus.support.language.deployment;
+
+import java.io.IOException;
+import java.io.StringWriter;
+import java.io.Writer;
+import java.util.Properties;
+import java.util.function.Consumer;
+
+import io.quarkus.builder.BuildChainBuilder;
+import io.quarkus.test.QuarkusUnitTest;
+import jakarta.inject.Inject;
+import org.apache.camel.CamelContext;
+import org.apache.camel.builder.RouteBuilder;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.asset.Asset;
+import org.jboss.shrinkwrap.api.asset.StringAsset;
+import org.jboss.shrinkwrap.api.spec.JavaArchive;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class PropertiesToSetTest {
+
+    @RegisterExtension
+    static final QuarkusUnitTest CONFIG = new QuarkusUnitTest()
+            .addBuildChainCustomizer(new Consumer<>() {
+                @Override
+                public void accept(BuildChainBuilder buildChainBuilder) {
+                    buildChainBuilder.addFinal(ExpressionExtractionResultBuildItem.class);
+                }
+            })
+            .setArchiveProducer(() -> ShrinkWrap.create(JavaArchive.class)
+                    .addAsResource(applicationProperties(), "application.properties"));
+
+    public static Asset applicationProperties() {
+        Writer writer = new StringWriter();
+
+        Properties props = new Properties();
+        props.setProperty("quarkus.camel.expression.on-build-time-analysis-failure", "fail");
+
+        try {
+            props.store(writer, "");
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+
+        return new StringAsset(writer.toString());
+    }
+
+    @Inject
+    CamelContext context;
+
+    @Test
+    void routeWithPropertiesToSet() {
+        assertEquals(1, context.getRoutesSize());
+    }
+
+    public static class Routes extends RouteBuilder {
+        @Override
+        public void configure() throws Exception {
+            from("timer:myTimer?period=1&repeatCount=1&delay=1")
+                    .log("body value is: ${body}");
+        }
+    }
+}

